### ISSUE TITLE
Initially dragging textures (PNG, JPG) from FileSystem into shader parameters lags

### ISSUE THREAD

**Issue by bikemurt** [godotengine/godot#88060]
### Tested versions

- Reproducible in 4.2.1 b09f793f5

### System information

Godot v4.2.1.stable - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3050 Ti Laptop GPU (NVIDIA; 31.0.15.5123) - 12th Gen Intel(R) Core(TM) i5-12500H (16 Threads)

### Issue description

After dragging a texture from FileSystem to a shader parameter for the first time, the lag was something like 10 seconds -I thought Godot was going to crash. If you drag and hold and wait long enough you can actually drop into the shader parameter but the UI is noticeably laggy during this portion.

After clearing the shader parameter and dragging it in again, there was no lag. So it only happens on the first time.

Happens with JPG, PNG

It does NOT happen with DDS, so I think this points to some sort of problem with the import pipeline.

### Steps to reproduce

To reproduce, copy/paste the shader below into a new ShaderMaterial, load the texture into the FileSystem, and try dragging the `pebbles_01.jpg` from the FileSystem into the Texture 2 parameter.

`shader_type spatial;

uniform sampler2D texture1;
uniform sampler2D texture2;
uniform sampler2D splatMap;

void fragment() {
	// Called for every pixel the material is visible on.
}
`

![pebbles_02](https://github.com/godotengine/godot/assets/23486102/65856d47-ae01-4305-8954-c89283c07d44)

### Minimal reproduction project (MRP)

[DragTextureIssue.zip](https://github.com/godotengine/godot/files/14195065/DragTextureIssue.zip)

**Comment by bikemurt** [comment]
I don't know enough about image formats and compression to have a really intelligible conversation about this - but I am wondering if the DDS format is fine because it's basically already in a format easily digestible for VRAM?

As per loading bar, no. The UI froze for about 10 seconds and then when the icon appeared to drop the texture in, it was lagging about a second behind my mouse movement.

Is there no way to trigger the VRAM compression upon first importing the file rather than when it is first used in the rendering pipeline?

**Comment by Calinou** [comment]
>  but I am wondering if the DDS format is fine because it's basically already in a format easily digestible for VRAM?

Yes – in fact, Godot doesn't even import DDS textures. It reads them directly instead.

**Comment by quackmarx** [comment]
Just linking a previous discussion on the topic: #78471

See my last comment on the discussion for arguments as to why this should be fixed.

> When a texture is first imported, we don't know in advance whether it'll be used in a 3D context yet. Importing all textures with VRAM compression in the hope that they will eventually be used in 3D would waste resources, as this won't happen with textures used in a 2D project or UI textures.

I think it's important we find a place to put this loading that is not during a UI/UX operation such as drag and drop. Perhaps after the drag/drop has concluded, sacrificing some of the preview functionality in the viewport?

> The alternative is to import the texture with lossless compression (so you can see it apply faster) and immediately import it with VRAM compression afterwards, but this will take more time than importing it directly with VRAM compression.

I would personally prefer this. I think a little but of extra load time is for me preferred to hanging up the editor mid-drag.

**Comment by Calinou** [comment]
VRAM compression should be greatly sped up with https://github.com/godotengine/godot/pull/95915 in 4.4, so this should be much less noticeable now.
